pkg/autodiscovery/providers/kube_endpoints_file.go

```diff
@@ -0,0 +1,310 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+// +build clusterchecks
+// +build kubeapiserver
+
+package providers
+
+import (
+	"context"
+	"fmt"
+	"sync"
+
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/common/utils"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/integration"
+	"github.com/DataDog/datadog-agent/pkg/autodiscovery/providers/names"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	"github.com/DataDog/datadog-agent/pkg/util/kubernetes/apiserver"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/equality"
+	listersv1 "k8s.io/client-go/listers/core/v1"
+	"k8s.io/client-go/tools/cache"
+)
+
+func init() {
+	RegisterProvider(names.KubeEndpointsFileRegisterName, NewKubeEndpointsFileConfigProvider)
+}
+
+type store struct {
+	sync.RWMutex
+	epConfigs map[string]*epConfig
+}
+
+func newStore() *store {
+	return &store{epConfigs: make(map[string]*epConfig)}
+}
+
+type epConfig struct {
+	templates     []integration.Config
+	ep            *v1.Endpoints
+	shouldCollect bool
+}
+
+func newEpConfig() *epConfig {
+	return &epConfig{
+		templates:     []integration.Config{},
+		shouldCollect: false,
+	}
+}
+
+// KubeEndpointsFileConfigProvider generates endpoints checks from check configurations defined in files.
+type KubeEndpointsFileConfigProvider struct {
+	sync.RWMutex
+	epLister listersv1.EndpointsLister
+	upToDate bool
+	store    *store
+}
+
+// NewKubeEndpointsFileConfigProvider returns a new KubeEndpointsFileConfigProvider
+func NewKubeEndpointsFileConfigProvider(config config.ConfigurationProviders) (ConfigProvider, error) {
```

Comment:
it seems that `config` is not used.

Reply:
removed the config argument. I can change the signature of the function in a separate PR.